src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -0,0 +1,116 @@
+import Boom from 'boom';
+import { get, omit } from 'lodash';
+
+import {
+  createFindQuery,
+  createFilterPath,
+  handleEsError,
+} from './lib';
+
+export class SavedObjectsClient {
+  constructor(kibanaIndex, callAdminCluster) {
+    this._kibanaIndex = kibanaIndex;
+    this._callAdminCluster = callAdminCluster;
+  }
+
+  async create(type, options = {}) {
+    const body = omit(options, 'id');
+
+    const response = await this._withKibanaIndex('index', {
+      type,
+      body
+    });
+
+    return Object.assign({
+      type: response._type,
+      id: response._id,
+      version: response._version
+    }, body);
+  }
+
+  async delete(type, id) {
+    const response = await this._withKibanaIndex('delete', {
+      type,
+      id,
+      refresh: 'wait_for'
+    });
+
+    if (get(response, 'found') === false) {
+      throw Boom.notFound();
+    }
+
+    return get(response, 'result') === 'deleted';
+  }
+
+  async find(options = {}) {
+    const {
+      search,
+      searchFields,
+      type,
+      fields,
+      perPage = 20,
+      page = 1,
+    } = options;
+
+    const esOptions = {
+      type,
+      filterPath: fields ? createFilterPath(fields) : undefined,
```

Comment:
I don't think this should be using `filterPath`, which is applied to the completed response before returning it to the client. We should almost certainly be use source filtering (see [`_sourceInclude`](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#api-search) param)

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-source-filtering.html